src/history.cxx

```diff
@@ -127,6 +127,12 @@ History History::deepcopy() const
 
 void History::set_data(double * input)
 {
+  if (store_) {
+    store_ = false;
+    storesize_ = 0;
+    delete [] storage_;
+    storage_ = nullptr;
+  }
```

Comment:
What's an `unset` method?

Having a method that just removes the association between a flat array and the history object makes no sense because it leaves the history method as invalid (you couldn't use it anymore).  Deleting that history object makes sense (and we have a deconstructor) as would making a copy of the history that now owned its storage (which we have).

Or are you thinking of some other use case?